[GStreamer][WebRTC] Notify ICE candidates also if there's no corresponding mid

philn opened the pull request:
#### fab65f9ffd62b26ec0c361b2c9cb04e5b5e56f43
<pre>
[GStreamer][WebRTC] Notify ICE candidates also if there&apos;s no corresponding mid
<a href="https://bugs.webkit.org/show_bug.cgi?id=252405">https://bugs.webkit.org/show_bug.cgi?id=252405</a>

Reviewed by Xabier Rodriguez-Calvar.

As bug title says. Also drive-by improving log reporting a bit.

* Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp:
(WebCore::GStreamerMediaEndpoint::setDescription):
(WebCore::GStreamerMediaEndpoint::onIceCandidate):

Canonical link: <a href="https://commits.webkit.org/260505@main">https://commits.webkit.org/260505@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7f189f0c75bb69bfd4862b3bda9ecfc54e285874

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108157 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17234 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41024 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117282 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116600 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/112043 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18645 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8524 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100363 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113925 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14061 "Too many flaky failures: editing/deleting/smart-delete-003.html, editing/deleting/smart-delete-paragraph-005.html, editing/pasteboard/smart-paste-003-trailing-whitespace.html, editing/pasteboard/smart-paste-005.html, editing/pasteboard/smart-paste-in-text-control.html, editing/pasteboard/smart-paste-paragraph-004.html, editing/selection/doubleclick-whitespace-img-crash.html, editing/selection/ios/select-text-under-hidden-subframe.html, editing/selection/ios/selection-handles-in-iframe.html, imported/w3c/web-platform-tests/html/semantics/popovers/popover-animated-hide-cleanup.tentative.html, platform/ipad/fast/viewport/meta-viewport-ignored.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97240 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41947 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95972 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28878 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/83617 "Found 1 new API test failure: /WebKitGTK/TestDownloads:/webkit/Downloads/local-file-error (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10106 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30226 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10829 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7127 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16255 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49817 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7262 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12410 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp:
```diff
@@ -1184,14 +1189,16 @@ void GStreamerMediaEndpoint::onIceCandidate(guint sdpMLineIndex, gchararray cand
-        if (!description)
-            return;
+        if (description) {
+            if (const auto* media = gst_sdp_message_get_media(description->sdp, sdpMLineIndex))
```
You can maybe collapse these two ifs and trust short-circuit evaluation.

philn replied:
Not sure to understand what you're suggesting here :)

calvaris replied:
`if (description && (const auto* media = gst_sdp_message_get_media(description->sdp, sdpMLineIndex))`

Though now that I think of it, maybe the compiler does not allow the variable declaration...

philn committed:
[GStreamer][WebRTC] Notify ICE candidates also if there's no corresponding mid
https://bugs.webkit.org/show_bug.cgi?id=252405

Reviewed by Xabier Rodriguez-Calvar.

As bug title says. Also drive-by improving log reporting a bit.

* Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp:
(WebCore::GStreamerMediaEndpoint::setDescription):
(WebCore::GStreamerMediaEndpoint::onIceCandidate):

Canonical link: https://commits.webkit.org/260505@main
